packages/openapi-generator/src/openapi-types.ts

```diff
@@ -153,7 +153,9 @@ export type OpenApiSchema =
   | OpenApiOneOfSchema
   | OpenApiAllOfSchema
   | OpenApiAnyOfSchema
-  | OpenApiNotSchema;
+  | OpenApiNotSchema
+) &
+  OpenApiAdditionalSchemaProperty;
```

Comment:
I have combined properties from interface `OpenApiAdditionalSchemaProperty `to the `OpenApiTypeSpecificProperty `(now renamed to `OpenApiSchemaProperties `). Since `deprecated` and `example` can be at both persisted schema level and object property level, and we are adding `typeProperties ` to both too, it made more sense to me to combine the interfaces into one.